### ISSUE TITLE
Darwin doesn't build using `nixpkgs-unstable`

### ISSUE THREAD

**Issue by malob** [LnL7/nix-darwin#701]
```shell
~/C/nix-darwin on master
❯ nix-build ./release.nix -I nixpkgs=channel:nixpkgs-unstable -I darwin=. -A manualHTML
warning: Nix search path entry '/nix/var/nix/profiles/per-user/root/channels' does not exist, ignoring
these 5 derivations will be built:
  /nix/store/dgcg2rh6gxq3z5vjf3bm5ddba265ad6b-options-db.xml.drv
  /nix/store/qn5cf5gppksjidq9h95cxra00cixjrfq-generated-docbook.drv
  /nix/store/fsl8dipccd21vm78m8cz59l6jlz0pa4r-darwin-manual-combined.drv
  /nix/store/wwza9wlw4wbz889d658ma8n54czhcsqa-manual-olinkdb.drv
  /nix/store/d06lnmif0lkvbfygzkyny9a0h8zn3l0s-darwin-manual-html.drv
building '/nix/store/dgcg2rh6gxq3z5vjf3bm5ddba265ad6b-options-db.xml.drv'...
intermediate.xml:3655: parser error : Opening and ending tag mismatch: name line 3653 and para
</para></nixos:option-description><para><emphasis>Type:</emphasis> path</para><p
       ^
intermediate.xml:3655: parser error : Opening and ending tag mismatch: name line 3653 and nixos:option-description
</para></nixos:option-description><para><emphasis>Type:</emphasis> path</para><p
                                  ^
intermediate.xml:3663: parser error : Opening and ending tag mismatch: para line 3652 and listitem
            </filename></member></simplelist></listitem></varlistentry><varliste
                                                        ^
intermediate.xml:3663: parser error : Opening and ending tag mismatch: option-description line 3652 and varlistentry
            </filename></member></simplelist></listitem></varlistentry><varliste
                                                                       ^
intermediate.xml:4154: namespace error : Failed to parse QName 'https:'
For the exhaustive list of settings, see <https://docs.hercules-ci.com/hercules-
                                                ^
intermediate.xml:4154: parser error : error parsing attribute name
For the exhaustive list of settings, see <https://docs.hercules-ci.com/hercules-
                                                ^
intermediate.xml:4154: parser error : attributes construct error
For the exhaustive list of settings, see <https://docs.hercules-ci.com/hercules-
                                                ^
intermediate.xml:4154: parser error : Couldn't find end of Start Tag https: line 4154
For the exhaustive list of settings, see <https://docs.hercules-ci.com/hercules-
                                                ^
intermediate.xml:4176: namespace error : Failed to parse QName 'https:'
The format is described on <https://docs.hercules-ci.com/hercules-ci-agent/binar
                                  ^
intermediate.xml:4176: parser error : error parsing attribute name
The format is described on <https://docs.hercules-ci.com/hercules-ci-agent/binar
                                  ^
intermediate.xml:4176: parser error : attributes construct error
The format is described on <https://docs.hercules-ci.com/hercules-ci-agent/binar
                                  ^
intermediate.xml:4176: parser error : Couldn't find end of Start Tag https: line 4176
The format is described on <https://docs.hercules-ci.com/hercules-ci-agent/binar
                                  ^
intermediate.xml:4184: namespace error : Failed to parse QName 'https:'
<https://hercules-ci.com/dashboard>.
       ^
intermediate.xml:4184: parser error : error parsing attribute name
<https://hercules-ci.com/dashboard>.
       ^
intermediate.xml:4184: parser error : attributes construct error
<https://hercules-ci.com/dashboard>.
       ^
intermediate.xml:4184: parser error : Couldn't find end of Start Tag https: line 4184
<https://hercules-ci.com/dashboard>.
       ^
intermediate.xml:4240: namespace error : Failed to parse QName 'https:'
The format is described on <https://docs.hercules-ci.com/hercules-ci-agent/secre
                                  ^
intermediate.xml:4240: parser error : error parsing attribute name
The format is described on <https://docs.hercules-ci.com/hercules-ci-agent/secre
                                  ^
intermediate.xml:4240: parser error : attributes construct error
The format is described on <https://docs.hercules-ci.com/hercules-ci-agent/secre
                                  ^
intermediate.xml:4240: parser error : Couldn't find end of Start Tag https: line 4240
The format is described on <https://docs.hercules-ci.com/hercules-ci-agent/secre
                                  ^
intermediate.xml:5930: parser error : Opening and ending tag mismatch: listitem line 3652 and variablelist
      </filename></member></simplelist></listitem></varlistentry></variablelist>
                                                                               ^
intermediate.xml:5930: parser error : Opening and ending tag mismatch: varlistentry line 3652 and appendix
lename></member></simplelist></listitem></varlistentry></variablelist></appendix
                                                                               ^
intermediate.xml:5932: parser error : Premature end of data in tag variablelist line 3

^
unable to parse intermediate.xml

**Comment by jonpulsifer** [comment]
for what it's worth, disabling documentation allows the build to complete (as of right now)

```nix
# items removed for brevity
darwinSystem {
  system = "aarch64-darwin";
  modules = [
    { documentation.enable = false; }
  ];
}
```

The documentation module defaults to enabled today https://github.com/LnL7/nix-darwin/blob/ed275afbbaad9b0670e2aeac3ae542595255d604/modules/documentation/default.nix#L57

**Comment by malob** [comment]
Thanks, yeah I did the same.

**Comment by billimek** [comment]
I'm experiencing the same error but not trying to use unstable (as far as I know).  This is a first-time attempt on this macbook.  Not sure if I should open a new issue?

* Installed nix for the first time via `curl -L https://nixos.org/nix/install | sh` on MacOS 13.4 (M1 Air)
* Ran basic nix-darwin build and install commands: `nix-build https://github.com/LnL7/nix-darwin/archive/master.tar.gz -A installer && ./result/bin/darwin-installer`

```shell
❯ nix-shell -p nix-info --run "nix-info -m"
 - system: `"aarch64-darwin"`
 - host os: `Darwin 22.5.0, macOS 13.4`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.16.1`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```

... And attempting to run the installer and associated errors

```shell
❯ ./result/bin/darwin-installer

Installing nix-darwin...

Would you like to edit the default configuration.nix before starting? [y/n] n
these 11 derivations will be built:
  /nix/store/vb8h370yabnkaa3ir5fldc4fwlk8rf8h-options-db.xml.drv
  /nix/store/q5249zn72x0wziq63cy6pn8w1pm5bqq3-generated-docbook.drv
  /nix/store/n2jlhkkcb5k42v42r8qggzn5rzn16zcp-darwin-manual-combined.drv
  /nix/store/04211z5bpg942r1pkik1c2y7azgr2ipi-manual-olinkdb.drv
  /nix/store/l8hvfi1phmwpm7pcdj13y8m6pc5xm622-darwin-manual-html.drv
  /nix/store/vpv3813bq7wnxs2i00km56di9fd42rq6-darwin-manpages.drv
  /nix/store/zw89kbmgh070q7q0q2f62g1l3hnppx0r-darwin-help.drv
  /nix/store/00qasypvddh80ch8yw70gspyjqkwjwpg-system-path.drv
  /nix/store/gnsvvc51inhmz8zd7p41x440zjmdbqij-etc.drv
  /nix/store/ldnl41a9gs9z0cxsgsxgk8hjciiiaxr1-system-applications.drv
  /nix/store/w7d8axhqyzvpdn8rw948vi1ynl5gvvaj-darwin-system-23.11pre496524.2c9ecd1f040+darwin4.0000000.drv
building '/nix/store/gnsvvc51inhmz8zd7p41x440zjmdbqij-etc.drv'...
building '/nix/store/vb8h370yabnkaa3ir5fldc4fwlk8rf8h-options-db.xml.drv'...
intermediate.xml:3655: parser error : Opening and ending tag mismatch: name line 3653 and para
</para></nixos:option-description><para><emphasis>Type:</emphasis> path</para><p
       ^
intermediate.xml:3655: parser error : Opening and ending tag mismatch: name line 3653 and nixos:option-description
</para></nixos:option-description><para><emphasis>Type:</emphasis> path</para><p
                                  ^
intermediate.xml:3663: parser error : Opening and ending tag mismatch: para line 3652 and listitem
            </filename></member></simplelist></listitem></varlistentry><varliste
                                                        ^
intermediate.xml:3663: parser error : Opening and ending tag mismatch: option-description line 3652 and varlistentry
            </filename></member></simplelist></listitem></varlistentry><varliste
                                                                       ^
intermediate.xml:4154: namespace error : Failed to parse QName 'https:'
For the exhaustive list of settings, see <https://docs.hercules-ci.com/hercules-
                                                ^
intermediate.xml:4154: parser error : error parsing attribute name
For the exhaustive list of settings, see <https://docs.hercules-ci.com/hercules-
                                                ^
intermediate.xml:4154: parser error : attributes construct error
For the exhaustive list of settings, see <https://docs.hercules-ci.com/hercules-
                                                ^
intermediate.xml:4154: parser error : Couldn't find end of Start Tag https: line 4154
For the exhaustive list of settings, see <https://docs.hercules-ci.com/hercules-
                                                ^
intermediate.xml:4176: namespace error : Failed to parse QName 'https:'
The format is described on <https://docs.hercules-ci.com/hercules-ci-agent/binar
                                  ^
intermediate.xml:4176: parser error : error parsing attribute name
The format is described on <https://docs.hercules-ci.com/hercules-ci-agent/binar
                                  ^
intermediate.xml:4176: parser error : attributes construct error
The format is described on <https://docs.hercules-ci.com/hercules-ci-agent/binar
                                  ^
intermediate.xml:4176: parser error : Couldn't find end of Start Tag https: line 4176
The format is described on <https://docs.hercules-ci.com/hercules-ci-agent/binar
                                  ^
intermediate.xml:4184: namespace error : Failed to parse QName 'https:'
<https://hercules-ci.com/dashboard>.
       ^
intermediate.xml:4184: parser error : error parsing attribute name
<https://hercules-ci.com/dashboard>.
       ^
intermediate.xml:4184: parser error : attributes construct error
<https://hercules-ci.com/dashboard>.
       ^
intermediate.xml:4184: parser error : Couldn't find end of Start Tag https: line 4184
<https://hercules-ci.com/dashboard>.
       ^
intermediate.xml:4240: namespace error : Failed to parse QName 'https:'
The format is described on <https://docs.hercules-ci.com/hercules-ci-agent/secre
                                  ^
intermediate.xml:4240: parser error : error parsing attribute name
The format is described on <https://docs.hercules-ci.com/hercules-ci-agent/secre
                                  ^
intermediate.xml:4240: parser error : attributes construct error
The format is described on <https://docs.hercules-ci.com/hercules-ci-agent/secre
                                  ^
intermediate.xml:4240: parser error : Couldn't find end of Start Tag https: line 4240
The format is described on <https://docs.hercules-ci.com/hercules-ci-agent/secre
                                  ^
intermediate.xml:5930: parser error : Opening and ending tag mismatch: listitem line 3652 and variablelist
      </filename></member></simplelist></listitem></varlistentry></variablelist>
                                                                               ^
intermediate.xml:5930: parser error : Opening and ending tag mismatch: varlistentry line 3652 and appendix
lename></member></simplelist></listitem></varlistentry></variablelist></appendix
                                                                               ^
intermediate.xml:5932: parser error : Premature end of data in tag variablelist line 3

^
unable to parse intermediate.xml
error: builder for '/nix/store/vb8h370yabnkaa3ir5fldc4fwlk8rf8h-options-db.xml.drv' failed with exit code 6;
       last 10 log lines:
       > intermediate.xml:5930: parser error : Opening and ending tag mismatch: listitem line 3652 and variablelist
       >       </filename></member></simplelist></listitem></varlistentry></variablelist>
       >                                                                                ^
       > intermediate.xml:5930: parser error : Opening and ending tag mismatch: varlistentry line 3652 and appendix
       > lename></member></simplelist></listitem></varlistentry></variablelist></appendix
       >                                                                                ^
       > intermediate.xml:5932: parser error : Premature end of data in tag variablelist line 3
       >
       > ^
       > unable to parse intermediate.xml
       For full logs, run 'nix-store -l /nix/store/vb8h370yabnkaa3ir5fldc4fwlk8rf8h-options-db.xml.drv'.
error: 1 dependencies of derivation '/nix/store/q5249zn72x0wziq63cy6pn8w1pm5bqq3-generated-docbook.drv' failed to build
error: 1 dependencies of derivation '/nix/store/n2jlhkkcb5k42v42r8qggzn5rzn16zcp-darwin-manual-combined.drv' failed to build
error: 1 dependencies of derivation '/nix/store/vpv3813bq7wnxs2i00km56di9fd42rq6-darwin-manpages.drv' failed to build
error: 1 dependencies of derivation '/nix/store/l8hvfi1phmwpm7pcdj13y8m6pc5xm622-darwin-manual-html.drv' failed to build
error: 1 dependencies of derivation '/nix/store/ldnl41a9gs9z0cxsgsxgk8hjciiiaxr1-system-applications.drv' failed to build
error: 1 dependencies of derivation '/nix/store/w7d8axhqyzvpdn8rw948vi1ynl5gvvaj-darwin-system-23.11pre496524.2c9ecd1f040+darwin4.0000000.drv' failed to build
```

It looks related to the documentation issue y'all mention but I'm not sure how to disable documentation in the standard generic configuration.nix that gets generated and used in the basic installer?

**Comment by YOUSIKI** [comment]
@billimek To disable docomentation in the basic installer, just add one line to your `.nixpkgs/darwin-configuration.nix` before continuing installation.
```nix
{ config, pkgs, ... }:

{
  documentation.enable = false;

  ...
}
```

**Comment by malob** [comment]
I think I found the commit in `nixpkgs` that is responsible: https://github.com/NixOS/nixpkgs/commit/d36f950d405369abca5144b14a201bc25c4a080a (PR: https://github.com/NixOS/nixpkgs/pull/237557)

```shell
~/C/nixpkgs on nixpkgs-unstable
❯ git bisect start HEAD 23.05
Bisecting: a merge base must be tested
[29cdfd50f06e966f7498a21d344a6385299f1177] Merge pull request #233234 from figsoda/abbr

~/C/nixpkgs (29cdfd5) (BISECTING) 
❯ git bisect run nix-build ../nix-darwin/release.nix -I nixpkgs=.  -I darwin=../nix-darwin -A manualHTML

[...]

d36f950d405369abca5144b14a201bc25c4a080a is the first bad commit
commit d36f950d405369abca5144b14a201bc25c4a080a
Author: pennae <github@quasiparticle.net>
Date:   Sun Jun 11 20:27:54 2023 +0200

    lib: turn *MD functions into aliases
    
    with docbook gone and MD the default these aren't needed any more. we
    can't remove them yet because there's thousands of uses, but maybe some
    day we can.

 lib/modules.nix | 19 +++++++------------
 lib/options.nix | 19 +++++++------------
 2 files changed, 14 insertions(+), 24 deletions(-)
bisect found first bad commit
```

**Comment by emilazy** [comment]
Is there any plan to convert `nmd` over to the same Markdown-based setup NixOS is moving to? It seems like a ticking time bomb to try to keep the DocBook chain working after upstream abandons it.

**Comment by emilazy** [comment]
Working on fixing this. Nixpkgs is amenable to reverting the alias change if it causes us too many issues, but I think we can just convert the DocBook option documentation now and be done with it.
